reorganize recipe card customization doc (#77)
miamtech/mealz-documentation: docs/android/features/recipe-card/customize-view/RecipeCard-customization.mdx

---
sidebar_position: 1
label: Recipe card customization
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import UseStringResource from '../../../_code/_utils/UseStringResource.mdx'

# RecipeCard customization

> RecipeCard is one of the most use components, it has three customization states: **success**, **loading** & **empty**

Reviewer: ```suggestion
RecipeCard is one of the most used components, it has three customization states: **success**, **loading** & **empty**
```

## customisation setting

If you want to fully customize this component we recommend copying the code into your Mealz template configuration.
You can remove the code you don't want to overwrite.

```kotlin
MiamTheme.Template {
    recipeCard {
        success {
            // Deprecated you should use catalog and shelf instead
            view = RecipeCardSuccessCustomView()
            catalog {
                view = RecipeCardCatalogSuccessCustomView()
            }
            shelf {
                view = RecipeCardShelfSuccessCustomView()
            }
        }
        loading {
            // Deprecated you should use catalog and shelf instead
            view = RecipeCardLoadingCustomView()
            catalog {
                view = RecipeCardCatalogLoadingCustomView()
            }
            shelf {
                view = RecipeCardShelfLoadingCustomView()
            }
        }
    }
}
```

## Success

> Recipe card success is one of those three states, it's visible when all data has been fetched without any error. 

Reviewer: ```suggestion
Recipe card success is one of those three states, it's visible when all of the data has been fetched without any error. 
```

> To create your own recipe card success template you create a class that implements `RecipeCardSuccess`

Reviewer: ```suggestion
To create your own recipe card success template you create a class that implements `RecipeCardSuccess`.
```

:::warning

> View temmplate is **Deprecated** you must use **catalog** and **shelf** instead

Reviewer: ```suggestion
View template is **Deprecated** you must use **catalog** and **shelf** instead
```

      val mealzRecipeId: String, // recipe id in mealz data base
      val recipePicture: String, 
      val recipePrice: Double,
      val recipeIngredients: Map<String,String>, // name of ingredient / ImageURL 
      val goToDetail: () -> Unit, // action to open recipe detail
      val guest: MutableStateFlow<Int>, 
      val isInCart: Boolean, // true if recipe is in cart
      val isSponsor: Boolean, //  true if recipe has a sponsor
      val sponsorLogo: String?,
      val likeIsEnabled: Boolean //  true is like feature has been enabled
){
      val pricePerServe : Double
           get() {
                val guest : Double =  if (guest.value == 0) 1.0 else guest.value.toDouble()
                return   recipePrice / guest
           }
 }

```

### Success ressources 

<UseStringResource localisationExample = {"Localisation.recipe.showDetails.localised"}/>

| Name | Ressource ID | Value Fr | Value Eng |
| - | - | - | - |
| showDetails | com_miam_recipe_is_in_cart | Voir le détail | Show details |
| add | com_miam_recipe_add | Ajouter les ingrédients | Add ingredients |
| perPerson | com_miam_my_meals_per_person | /pers.| /pers. |

## RecipeCard Loading

> As for Success state basic Laoding view has been **deprecated** you must use catalog and shelf loading.

Reviewer: ```suggestion
As for the Success state, the basic Loading view has been **deprecated** you must use catalog and shelf loading.
```

<Tabs defaultValue="empty_example" groupId="code-example"
values={[
{ label: 'RecipeCardLoadingCustomView.kt', value: 'empty_example' },
{ label: 'example', value: 'full_example' },
]}>
<TabItem value="empty_example">
```kotlin
import ai.mealz.sdk.components.recipeCard.loading.RecipeCardLoading

class RecipeCardLoadingCustomView: RecipeCardLoading {

    @Composable
    override fun Content() {
        // Your custom design here 
    }
}
```
</TabItem>
<TabItem value="full_example">
```koltin
import ai.mealz.sdk.components.recipeCard.loading.RecipeCardLoading

class RecipeCardLoadingCustomView: RecipeCardLoading {

    @Composable
    override fun Content() {
        Column(Modifier.fillMaxWidth().padding(vertical= 16.dp), horizontalAlignment = Alignment.CenterHorizontally){
            CircularProgressIndicator()
        }
    }
}
```
</TabItem>
</Tabs>

###  Loading catalog

> You can choose to have to differents design for the recipe card in our catalog and in your shelf.

Reviewer: ```suggestion
You can choose to have to different designs for the recipe card in our catalog and in your shelf.
```
